pkgs/applications/science/logic/stp/default.nix

```diff
@@ -26,7 +37,13 @@ stdenv.mkDerivation rec {
     sed -e '1i #include <cstdint>' -i include/stp/AST/ASTNode.h
   '';
 
-  buildInputs = [ boost zlib minisat cryptominisat python3 ];
+  buildInputs = [
+    boost
+    zlib
+    minisat
+    cryptominisat
+    (python3Packages.python.withPackages (p: [ p.setuptools ]))
```

Comment:
`distutils` is a native build input to determine the installation destination so ideally the Python binding's CMake file should be patched to skip it altogether.

Reply:
Thanks for the confirmation. I ended up doing this for z3: https://github.com/NixOS/nixpkgs/pull/327052/files

I'll do it for this too.

Reply:
IMHO with `python_install_dir` passed to CMake, we should be able to skip firing up Python:

https://github.com/stp/stp/blob/2.3.3/bindings/python/stp/CMakeLists.txt#L47-L61

The same for Z3 I suppose.

Reply:
Z3 is even slightly easier to control from our side because we can just set CMAKE_INSTALL_PYTHON_PKG_DIR: 

https://github.com/Z3Prover/z3/pull/5729/files#diff-03fa5bcf870844dbf2dc0a66027112e4f0ff0dd5c5cd5a7f946b1567b240be37L98-L121

We'd probably need a patch for STP, it seems like they run Python unconditionally of whether you have the variable set.

Reply:
I'm liking just pulling in this patch better: https://github.com/stp/stp/pull/450/files